datadog_checks_base/datadog_checks/base/checks/kube_leader/mixins.py

```diff
@@ -0,0 +1,106 @@
+# (C) Datadog, Inc. 2018
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+
+from kubernetes import client, config
+
+try:
+    import datadog_agent
+except ImportError:
+    from ...stubs import datadog_agent
+
+from .. import AgentCheck
+from .record import ElectionRecord
+
+# Known names of the leader election annotation,
+# will be tried in the order of the list
+ELECTION_ANNOTATION_NAMES = ["control-plane.alpha.kubernetes.io/leader"]
+
+
+class KubeLeaderElectionMixin(object):
+    """
+    This mixin uses the facilities of the AgentCheck class
+    """
+
+    def check_election_status(self, config):
```

Comment:
If so then maybe make default `check` method that just calls this like in the example?